scarb/src/core/lockfile.rs

```diff
@@ -0,0 +1,244 @@
+#![allow(dead_code)]
+
+use anyhow::{anyhow, Context, Result};
+use camino::Utf8Path;
+use semver::Version;
+use serde::{Deserialize, Serialize};
+use serde_repr::{Deserialize_repr, Serialize_repr};
+use std::collections::BTreeSet;
+use toml_edit::Document;
+
+use crate::core::{PackageId, PackageName, Resolve, SourceId};
+use crate::internal::fsx;
+
+const HEADER: &str = "# Code generated by scarb DO NOT EDIT.";
+
+#[derive(
+    Default, PartialEq, Eq, Clone, Copy, Debug, PartialOrd, Ord, Serialize_repr, Deserialize_repr,
+)]
+#[repr(u8)]
```

Comment:
https://serde.rs/enum-number.html

Reply:
can you also apply this to Cairo data structures? Cairo is in prerelease phase, we can make breaking changes yet <3